src/state/apis/nft/parsers/alchemy.ts

```diff
@@ -69,50 +71,68 @@ export const parseAlchemyNftContractToCollectionItem = (
   }
 }
 
-export const parseAlchemyOwnedNftToNftItem = (
-  alchemyOwnedNft: OwnedNft,
+export const parseAlchemyNftToNftItem = async (
+  alchemyNft: OwnedNft | Nft,
   chainId: ChainId,
-): NftItemWithCollection => {
+): Promise<NftItemWithCollection> => {
   const collectionId = toAssetId({
-    assetReference: alchemyOwnedNft.contract.address,
-    assetNamespace: alchemyOwnedNft.contract.tokenType.toLowerCase() as AssetNamespace,
+    assetReference: alchemyNft.contract.address,
+    assetNamespace: alchemyNft.contract.tokenType.toLowerCase() as AssetNamespace,
     chainId,
   })
-  const socialLinks = makeSocialLinks(alchemyOwnedNft.contract.openSea)
+  const socialLinks = makeSocialLinks(alchemyNft.contract.openSea)
 
   const nftCollection = {
     assetId: collectionId,
     chainId,
-    name:
-      alchemyOwnedNft.contract.name ||
-      alchemyOwnedNft.contract.openSea?.collectionName ||
-      'Collection',
+    name: alchemyNft.contract.name || alchemyNft.contract.openSea?.collectionName || 'Collection',
     floorPrice: '', // Seemingly unreliable
     openseaId: '', // The Alchemy NFT data does not have an openseaId
-    description: alchemyOwnedNft.contract.openSea?.description ?? '',
+    description: alchemyNft.contract.openSea?.description ?? '',
     socialLinks,
   }
 
+  // If we have an IPNS gateway metadata URL, it means unchained can get a fresh media URL
+  // Which allows us to circumvent Alchemy refresh working for Ethereum only
+  // Notes:
+  // - We're only able to get fresh meta from unchained for IPNS URLs, not IPFS ones
+  // - This hasn't been tested on Optimism, thus we only support this refresh for Polygon for now
```

Comment:
What I'd suggest if I was nitpicky
```suggestion
  // - This hasn't been tested on Optimism, hence we only support this refresh for Polygon for now
```

Reply:
89f45a704